geode-core/src/main/java/org/apache/geode/internal/cache/snapshot/RegionSnapshotServiceImpl.java

```diff
@@ -342,6 +342,7 @@ private void exportOnMember(File snapshot, SnapshotFormat format, SnapshotOption
 
     long count = 0;
     long start = CachePerfStats.getStatTime();
+    snapshot.getParentFile().mkdirs();
```

Comment:
Can getParentFile() return null...How about handling failure during mkdirs()...

Reply:
You are correct: if only a relative route to a specific file is provided (which would be an error condition in this case), getParentFile() will return null. I can handle that case more gracefully.